libminifi/src/core/Connectable.cpp

```diff
@@ -85,7 +69,7 @@ bool Connectable::isSupportedRelationship(const core::Relationship &relationship
   }
 }
 
-bool Connectable::setAutoTerminatedRelationships(const std::set<Relationship> &relationships) {
+bool Connectable::setAutoTerminatedRelationships(const std::vector<Relationship>& relationships) {
```

Comment:
shouldn't we turn this into a template as well? (as `setSupportedProperties`)

Reply:
I changed `setAutoTerminatedRelationships` to a template in 529f7014bdaf548ebfd5198cd78cdd5ff9b5b136

but then I realized it doesn't need to be a template, and changed `setSupportedProperties`, `setSupportedRelationships` and `setAutoTerminatedRelationships` back to functions, with a `gsl::span` argument type, in cb2e0defeb0d19539590fe82cd6eb69e466ca2dc

let me know if you don't like cb2e0defeb0d19539590fe82cd6eb69e466ca2dc and want it changed back

Reply:
I'm fine with `gsl::span` my only concern is from the caller's side we injected a number of `std::array` calls where previously we used an initializer list, do we need them or can we drop them now that the methods are not templates?

Reply:
Unfortunately, initializer lists don't convert implicitly to `gsl::span`.  Nevertheless, we used to create a `std::vector` (with elements on the heap) and now we only create a `std::array` (with elements on the stack).

Reply:
You can overload them to accept both gsl::span and std::array, if you want.

Reply:
I prefer `some_function(std::array{foo, bar})` to `some_function({foo, bar})`; I think it's clearer and it's also slightly more efficient.  The only downside is a few more characters, but I don't think that's a problem.